QuizApp/Quiz/DataSource/Local/QuizDatabaseDataSource.swift

```diff
@@ -0,0 +1,62 @@
+import Foundation
+import RealmSwift
+
+protocol QuizDatabaseDataSourceProtocol {
+
+    func getAllQuizzes() -> [QuizDataObject]
+
+    func getQuizzes(for category: String) -> [QuizDataObject]
+
+    func saveQuizzes(quizzes: [QuizDataObject])
+
+}
+
+class QuizDatabaseDataSource: QuizDatabaseDataSourceProtocol {
+
+    private let realmQueue = DispatchQueue(label: "thread.realm")
+
+    private var realm: Realm?
+
+    init() {
+        instantiateRealm()
+    }
+
+    func getAllQuizzes() -> [QuizDataObject] {
+        guard let realm = realm else { return [] }
+
+        return Array(realm.objects(QuizDataObject.self))
```

Comment:
Why is fetching not done on realmQueue?

Reply:
Because function ```realm.objects(...)``` uses the queue sent in realm init ```realm = try Realm(queue: self.realmQueue)``` to fetch the data